### DISCUSSION TITLE
Bug: ignorePatterns is ignored

### DISCUSSION THREAD

**Discussion by its-dibo** [eslint/eslint#18244] [Q&A]
### Environment

Node version: 
npm version: 
Local ESLint version: 
Global ESLint version: 
Operating System: 

### What parser are you using?

Default (Espree)

### What did you do?

as per the doce `.eslintignore` is being ignored if `--ignore-path .gitignore` is used, which is logical
but in addition to these rules in .gitignore, we often need to apply some patterns, particularly for Eslint, I guess that `ignorePatterns[]` should be taken into consideration and take precedence over both .gitignore or .eslintignore.

what happens is that providing `--ignore-path` causes that `ignorePatterns[]` to be not used

### What did you expect to happen?

`gitignore[]` should take precedence over .gitignore

### What actually happened?

`gitignore[]` is being ignored if .gitignore is used

### Link to Minimal Reproducible Example

.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
`.eslintignore` and `ignorePatterns` will be no longer suppported in the the new config.  and for compat issue, we won't make changes to the eslintrc (deprecated).

https://eslint.org/docs/next/use/configure/configuration-files

**Reply by its-dibo** [reply]
`ignorePatterns[]` is replaced with [`ignore[]`](https://eslint.org/docs/next/use/configure/configuration-files#specifying-files-and-ignores), correct?

**Reply by aladdin-add** [reply]
👍 

global-ignore: https://eslint.org/docs/next/use/configure/configuration-files#globally-ignoring-files-with-ignores
